test/index.js

```diff
@@ -424,45 +469,75 @@ describe('simple-keyring', function () {
-        'hex',
-      );
-      const privKeyHex = ethUtil.bufferToHex(privateKey);
-      const message = 'Hello world!';
-      const encryptedMessage = encrypt({
-        publicKey: getEncryptionPublicKey(privateKey),
-        data: message,
-        version: 'x25519-xsalsa20-poly1305',
-      });
-
+    const address = '0xbe93f9bacbcffc8ee6663f2647917ed7a20a57bb';
+    const privateKey = Buffer.from(
+      '6969696969696969696969696969696969696969696969696969696969696969',
+      'hex',
+    );
+    const privKeyHex = ethUtil.bufferToHex(privateKey);
+    const message = 'Hello world!';
+    const encryptedMessage = encrypt({
+      publicKey: getEncryptionPublicKey(privateKey),
+      data: message,
+      version: 'x25519-xsalsa20-poly1305',
+    });
+    beforeEach(async function () {
       await keyring.deserialize([privKeyHex]);
+    });
+
+    it('returns the expected value', async function () {
       const decryptedMessage = await keyring.decryptMessage(
         address,
         encryptedMessage,
       );
       expect(message).toBe(decryptedMessage);
     });
+
+    it('throw error if account is not present', async function () {
+      await expect(
+        keyring.decryptMessage(invalidAddress, encryptedMessage),
+      ).rejects.toThrow('Simple Keyring - Unable to find matching address.');
+    });
+
+    it('throw error if wrong encrypted data object is passed', async function () {
+      await expect(keyring.decryptMessage(address, {})).rejects.toThrow(
+        'Encryption type/version not supported.',
+      );
+    });
   });
 
   describe('#encryptionPublicKey', function () {
-    it('returns the expected value', async function () {
-      const address = '0xbe93f9bacbcffc8ee6663f2647917ed7a20a57bb';
-      const privateKey = Buffer.from(
-        '6969696969696969696969696969696969696969696969696969696969696969',
-        'hex',
-      );
-      const publicKey = 'GxuMqoE2oHsZzcQtv/WMNB3gCH2P6uzynuwO1P0MM1U=';
-      const privKeyHex = ethUtil.bufferToHex(privateKey);
+    const address = '0xbe93f9bacbcffc8ee6663f2647917ed7a20a57bb';
+    const privateKey = Buffer.from(
+      '6969696969696969696969696969696969696969696969696969696969696969',
+      'hex',
+    );
+    const publicKey = 'GxuMqoE2oHsZzcQtv/WMNB3gCH2P6uzynuwO1P0MM1U=';
+    const privKeyHex = ethUtil.bufferToHex(privateKey);
+
+    beforeEach(async function () {
       await keyring.deserialize([privKeyHex]);
+    });
+
+    it('returns the expected value', async function () {
       const encryptionPublicKey = await keyring.getEncryptionPublicKey(
         address,
         privateKey,
       );
       expect(publicKey).toBe(encryptionPublicKey);
     });
+
+    it('throw error if address is blank', async function () {
+      await expect(
+        keyring.getEncryptionPublicKey('', privateKey),
+      ).rejects.toThrow('Must specify address.');
+    });
+
+    it('throw error if address is invalid', async function () {
```

Comment:
same point here. I think its confusing to say that the address is invalid. Rather it's not on the keyring.

Reply:
Updated 👍